src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/JiraDeploymentInfoSenderImpl.java

```diff
@@ -129,8 +135,16 @@ public JiraSendInfoResponse sendDeploymentInfo(final JiraDeploymentInfoRequest r
             return JiraCommonResponse.failureAccessToken(resolvedSiteConfig);
         }
 
+        if (enableGate && !Objects.equals(deploymentState, State.PENDING.value)) {
+            return JiraDeploymentInfoResponse.failureEnableGateWrongDeploymentDtate(
+                    deploymentState);
+        }
```

Comment:
We also return an error from the API in this situation where the state is not compatible with the command.

Does this error handling prevent the user from setting up the bad configuration?

If not, it may be better to use the API response to avoid coupling changes in the API to changes in the plugin application code. For example, if we allow the `initiate_deployment_gating` command with a different deployment state other than pending (for some reason) you'd have to change this code. Can we avoid that?